### PR TITLE
Added type specific None overloads

### PR DESCRIPTION
**Pull request by droyad**
After some more testing, the switch statement in the None method is still pretty expensive. Adding the specific overloads however makes improves things markedly. This also means that the .NET 5.0 implementation won't be as good, so we should keep these overloads.

### CODE REVIEW

**Review by paulegradie — COMMENTED** [review]
Good idea to do the perf tests on this. 

So for this we've got comments in the repo history - would it be worth writing a roslyn analzyer for this as well? Especially if we're going to use these past .net 5.0

File: source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs [hunk 1]
```diff
@@ -21,7 +22,6 @@ public static HashSet<T> ToHashSet<T>(this IEnumerable<T> source, IEqualityCompa
         }
 #endif
 
-#if NET40 || NET452 || NETSTANDARD2_0 || NETCOREAPP3_1
         // In .NET 5.0 Enumerable.Any() calls Count/Length so these are no longer needed
```
Comment (paulegradie) [inline source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs:25]: This comment should maybe be updated if we're going to keep these (all?) through 5.0.

Does that implementation use switch statements? How did you figure 5.0 will be less performant?

File: source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs [hunk 2]
```diff
@@ -46,39 +46,48 @@ public static bool Any<T>(this T[] array)
 
         public static bool Any<T>(this HashSet<T> list)
             => list.Count > 0;
-#endif
 
-#if NET452 || NETSTANDARD2_0 || NETCOREAPP3_1
+#if !NET40
```
[the remark this answers is not in the document]
Comment (droyad) [reply source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs:50]: I separated them so that the `Any` is grouped and `None` is grouped, so when we drop `NET40` it won't need to change to be clean

File: source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs (hunk 1 above)
Comment (droyad) [reply source/Octopus.CoreUtilities/Extensions/EnumerableExtensions.cs:25]: It uses a few if/else blocks, which should be fast as they work on types, but I found the slowness with the switch below and I don't think it harms keeping these.
https://github.com/stephentoub/corefx/blob/f47fce34d13d54b299dac49ea97647edf805041e/src/System.Linq/src/System/Linq/AnyAll.cs#L19
